.github/workflows/test_clients.yaml

```diff
@@ -12,27 +12,36 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-name: Validate Schema Against Clients
-on: [push, pull_request]
+name: Client Smoke Tests
+on:
+  pull_request:
+    paths:
+      - 'proto/**'
+      - 'third_party/**'
+      - 'testdata/**'
+      - '.github/workflows/test_clients.yaml'
```

Comment:
Great to see more efficiency in CI & human attention around this. I wonder if this kind of targeting will make it difficult for folks to understand which checks to expect for any given PR. I think that's addressable with contributor docs about the testing process.

I'm concerned that if there are delays in the GHA runner to post that a particular check is under-way, we could have a case where a PR looks green and a rapid turnaround review could lead to a premature merge because the people involved have trouble with the mental model of which checks to expect. I'm not sure if I've seen this particular class of problem lately though, so maybe the GHA infra has addressed it?

Reply:
I think your concern may be somewhat mitigated by ensuring we have required status checks. I also was exploring having diagrams or providing a similar table to what I have in the design doc

Reply:
Can you review .github/workflows/WORKFLOWS.md and see if this helps orient contributors?